src/napari_micromanager/_gui_objects/_toolbar.py

```diff
@@ -0,0 +1,368 @@
+        wdg.layout().setContentsMargins(5, 0, 5, 0)
+        wdg.layout().setSpacing(3)
+        wdg.setStyleSheet("border: 0px;")
+
+        mda = self._make_plugin_button("MDA", "MultiDimensional Acquisition")
+        explorer = self._make_plugin_button(
+            "Explorer", "MultiDimensional Grid Acquisition"
+        )
+        wdg.layout().addWidget(mda)
+        wdg.layout().addWidget(explorer)
+
+        plgs_toolbar.addWidget(wdg)
+
+        return plgs_toolbar
+
+    def _make_plugin_button(self, btn_text: str, tooltip: str) -> QPushButton:
+        """Create the QPushButton for the plugins QToolBar."""
+        btn = QPushButton(text=btn_text)
+        btn.setToolTip(tooltip)
+        btn.setMinimumHeight(TOOL_SIZE)
+        btn.setWhatsThis(btn_text)
+        btn.clicked.connect(self._show_dock_widget)
+        return btn
+
+    def _show_dock_widget(self, key: str = "") -> None:
+        """Look up widget class in DOCK_WIDGETS and add/create or show/raise.
+
+        `key` must be a key in the DOCK_WIDGETS dict or a `str` stored in
+        the `whatsThis` property of a `sender` `QPushButton`.
+        """
+        if not key:
+            # using QPushButton.whatsThis() property to get the key.
+            btn = cast(QPushButton, self.sender())
+            key = btn.whatsThis()
+
+        if key in self._dock_widgets:
+            # already exists
+            dock_wdg = self._dock_widgets[key]
+            dock_wdg.show()
+            dock_wdg.raise_()
+        else:
+            # creating it for the first time
+            # sourcery skip: extract-method
+            try:
+                wdg_cls = DOCK_WIDGETS[key][0]
+            except KeyError as e:
+                raise KeyError(
+                    "Not a recognized dock widget key. "
+                    f"Must be one of {list(DOCK_WIDGETS)} "
+                    " or the `whatsThis` property of a `sender` `QPushButton`."
+                ) from e
+            wdg = wdg_cls(parent=self, mmcore=self._mmc)
+
+            if isinstance(wdg, PropertyBrowser):
+                wdg.setSizePolicy(
+                    QSizePolicy.Policy.Expanding, QSizePolicy.Policy.Expanding
+                )
+                wdg._prop_table.setVerticalScrollBarPolicy(
+                    Qt.ScrollBarPolicy.ScrollBarAlwaysOff
+                )
+
+            dock_wdg = self._add_dock_widget(wdg, key, tabify=True)
+            self._dock_widgets[key] = dock_wdg
+
+    def _add_dock_widget(
+        self, widget: QWidget, name: str, floating: bool = False, tabify: bool = False
+    ) -> QtViewerDockWidget:
+        """Add a docked widget using napari's add_dock_widget."""
+        dock_wdg = self.viewer.window.add_dock_widget(
+            widget,
+            name=name,
+            area="right",
+            allowed_areas=["left", "right"],
+            tabify=tabify,
+        )
+        dock_wdg.setFloating(floating)
+        dock_wdg._close_btn = False
```

Comment:
```suggestion
        try:
            dock_wdg._close_btn = False
        except AttributeError:
             pass
```

anytime you access a private attribute of an external dependency like this, you should assume it will disappear someday and try to make your code still work